src/pages/ProductDetail/index.tsx

```diff
@@ -1,27 +1,24 @@
+import { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
 import { Product as ProductType } from '../../types/product';
 import ProductList from '../../components/ProductList/ProductList';
 import styles from './productDetail.module.css';
 import MainLayout from '../../layouts/MainLayout';
-import { useParams } from 'react-router-dom';
-import { useEffect, useState } from 'react';
 import Loading from '../../components/Loading';
 import { default as styleUtils } from '../../styles/modules/utils.module.css';
 import { PRODUCTS_MOCK as mockProducts } from '../../mock/products'; // Import mock data
 import { ProductAPIService } from '../../services/ProductAPIService';
 import Product from '../../components/Product/Product';
-
-const productsPerPage = 8;
-
-const useMockData = import.meta.env.USE_MOCK_FOR_API_FAIL;
+import { useProducts } from '../../hook/useProducts'; // Import the custom hook
 
 const ProductDetail = () => {
-  const { id } = useParams(); // Assuming id is passed as a route parameter
+  const { id } = useParams<{ id: string }>();
 
   const [product, setProduct] = useState<ProductType | null>(null);
-  const [products, setProducts] = useState<ProductType[]>([]);
   const [isLoading, setIsLoading] = useState(false);
-  const [currentPage, setCurrentPage] = useState(1);
-  const [hasMore, setHasMore] = useState(true);
+
+  // Use the useProducts hook for related products
+  const { products, isLoading: isRelatedProductsLoading, hasMore, loadMore } = useProducts(product?.category || '');
 
   const fetchProductById = async (productId: string) => {
```

Comment:
Can we do the same as `useProducts` create another custom hook for product by id

Reply:
Fixed